plugin/FormPost.php

```diff
@@ -227,6 +231,42 @@ private function buildFormData(DataFormConfiguration $pFormConfig, $formNo): For
 		return $pFormData;
 	}
 
+	/**
+	 * @param int $formId
+	 * @param array $hiddenFields
+	 *
+	 * @return void
+	 * @throws DependencyException
+	 * @throws NotFoundException
+	 * @throws UnknownFieldException
+	 */
+
+	private function readDefaultValueForHiddenFieldsByFormId(int $formId, array $hiddenFields)
+	{
+		$pContainerBuilder = new ContainerBuilder;
+		$pContainerBuilder->addDefinitions(ONOFFICE_DI_CONFIG_PATH);
+		$pContainer = $pContainerBuilder->build();
+
+		/** @var DefaultValueModelToOutputConverter $pDefaultValueRead */
+		$pDefaultValueRead = $pContainer->get(DefaultValueModelToOutputConverter::class);
+
+		$fields = [];
+		foreach ($hiddenFields as $field) {
+			$fields[] = $this->_pFieldsCollection->getFieldByKeyUnsafe($field);
+		}
+
+		$values = [];
+		foreach (array_chunk($fields, 100) as $fields) {
+			$pDefaultFields = $pDefaultValueRead->getConvertedMultiFields($formId, $fields);
+			if (count($pDefaultFields)) $values = array_merge($values, $pDefaultFields);
+		}
+
+		array_walk($values, function($value, $key) {
+			if ($value === "0" || !empty($value)) {
+				$_POST[$key] = $value;
+			}
+		});
```

Comment:
<img width="1371" alt="form" src="https://github.com/user-attachments/assets/875903f0-16f9-4fc5-beaa-f991d4721033">

**Issues:** 
- If a field has a default value set, we can remove it from the HTML but it will still be submitted.
- Changing the value in HTML can’t overwrite the default value set.
- Hidden checkboxes with a default value "y" submit "1" which is interpreted as "n".

**Fix:**
_plugin/FormPost.php_

```
private function readDefaultValueForHiddenFieldsByFormId(int $formId, array $hiddenFields)
...
array_walk($values, function($value, $key) {
	if ($value === "0" || !empty($value) && !isset($_POST[$key])) {
		$_POST[$key] = $value;
	}
});
```
We need `!isset($_POST[$key])` to make sure we can overwrite the default value set in the plugin